Document `cuda::maximum`, `cuda::minimum` (#3883)
NVIDIA/cccl: docs/libcudacxx/extended_api/functional/maximum_minimum.rst

.. _libcudacxx-extended-api-functional-maximum-minimum:

``cuda::maximum`` and ``cuda::minimum``
=======================================

.. code:: cuda

    template <typename T>
    struct maximum {
        [[nodiscard]] __host__ __device__ inline
        T operator()(T a, T b) const;
    };

    template <>
    struct maximum<void> {
        template <typename T1, typename T2>
        [[nodiscard]] __host__ __device__ inline
        cuda::std::common_type_t<T1, T2> operator()(T1 a, T2 b) const;
    };

    template <typename T>
    struct minimum {
        [[nodiscard]] __host__ __device__ inline
        T operator()(T a, T b) const;
    };

    template <>
    struct minimum<void> {
        template <typename T1, typename T2>
        [[nodiscard]] __host__ __device__ inline
        cuda::std::common_type_t<T1, T2> operator()(T1 a, T2 b) const;
    };

> Function objects for performing maximum and minimum. The functions behave as ``noexcept`` when the comparison between the values is also ``noexcept``.

Reviewer: ```suggestion
Function objects for performing maximum and minimum. The functions are ``noexcept`` when the comparison between the values is also ``noexcept``.
```
